src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs

```diff
@@ -169,37 +148,60 @@ void Run (DirectoryAssemblyResolver res)
 			var java    = new Dictionary<string, TypeDefinition> ();
 			var acw_map = new StreamWriter (temp_map_file);
 
+			var managedConflicts = new Dictionary<string, List<string>> ();
+			var javaConflicts    = new Dictionary<string, List<string>> ();
+
 			foreach (var type in java_types) {
 				string managedKey = type.FullName.Replace ('/', '.');
 				string javaKey    = JavaNativeTypeManager.ToJniName (type).Replace ('/', '.');
 
 				acw_map.WriteLine ("{0};{1}", type.GetPartialAssemblyQualifiedName (), javaKey);
 
 				TypeDefinition conflict;
+				bool hasConflict = false;
 				if (managed.TryGetValue (managedKey, out conflict)) {
-					Log.LogWarning (
-							"Duplicate managed type found! Mappings between managed types and Java types must be unique. " +
-							"First Type: '{0}'; Second Type: '{1}'.",
-							conflict.GetAssemblyQualifiedName (),
-							type.GetAssemblyQualifiedName ());
-					Log.LogWarning (
-							"References to the type '{0}' will refer to '{1}'.",
-							managedKey, conflict.GetAssemblyQualifiedName ());
-					continue;
+					if (!managedConflicts.ContainsKey (managedKey))
+						managedConflicts [managedKey] = new List<string> { conflict.GetAssemblyName () };
+					managedConflicts [managedKey].Add (type.GetAssemblyName ());
```

Comment:
This is hitting the `Dictionary` three times. You could do something like this: 
```csharp
//test this, I just typed it:
if (!managedConflicts.TryGetValue (managedKey, out var list)) {
    managedConflicts.Add (list = new List<string> ());
}
list.Add (type.GetAssemblyName ());
```
`ContainsKey` and `[]` each have to do a lookup in the dictionary. This example would lookup once.

Reply:
Nice idea! I'll update the PR.